dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java

```diff
@@ -274,22 +287,34 @@ private void validateTrackedEntityAttributeFilters(
     }
   }
 
-  private void validateGlobalSearchParameters(TrackedEntityQueryParams params, UserDetails user)
-      throws IllegalQueryException {
-    if (!isLocalSearch(params, user)) {
+  private void validatePagination(PageParams pageParams) throws BadRequestException {
+    if (pageParams == null) {
+      return;
+    }
 
+    int systemMaxPageSize = systemSettingsService.getCurrentSettings().getTrackedEntityMaxLimit();
+    if (systemMaxPageSize > 0 && pageParams.getPageSize() > systemMaxPageSize) {
+      throw new BadRequestException(
+          String.format(
+              "Invalid page size: %d. It must not exceed the system limit of KeyTrackedEntityMaxLimit %d.",
+              pageParams.getPageSize(), systemMaxPageSize));
+    }
+  }
+
+  private void validateSearchOutsideCaptureScopeParameters(
+      TrackedEntityQueryParams params, UserDetails user) throws IllegalQueryException {
+    if (!isSearchInCaptureScope(params, user)) {
       if (params.hasFilters()) {
         List<UID> searchableAttributeIds = getSearchableAttributeIds(params);
         validateSearchableAttributes(params, searchableAttributeIds);
       }
 
-      int maxTeLimit = getMaxTeLimit(params);
-      if (maxTeLimit > 0
-          && trackedEntityStore.getTrackedEntityCountWithMaxTrackedEntityLimit(params)
-              > maxTeLimit) {
-        throw new IllegalQueryException("maxteicountreached");
+      if (!params.hasTrackedEntities()) {
+        validateMinAttributesToSearch(params);
```

Comment:
if you then apply the above suggestion you can move these guards inside the targeted validation functions